substrate/frame/referenda/src/lib.rs

```diff
@@ -225,8 +226,11 @@ pub mod pallet {
 	#[pallet::extra_constants]
 	impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		#[pallet::constant_name(Tracks)]
-		fn tracks() -> Vec<Track<TrackIdOf<T, I>, BalanceOf<T, I>, BlockNumberFor<T, I>>> {
-			T::Tracks::tracks().map(|t| t.into_owned()).collect()
+		fn tracks() -> Vec<(TrackIdOf<T, I>, TrackInfo<BalanceOf<T, I>, BlockNumberFor<T, I>>)> {
```

Comment:
As @gui1117 already said, this is a hack. I'm okay with keeping it for now. However, we should also directly add a `view` function and then create an issue to remove this constant.

Reply:
Of course! I understand this is a hack more than a permanent solution.

A more permanent solution is using a proper `BoundedVec`. But to do so, we either need a way to statically initialize a variable that also contains `BoudedVec` fields (that use allocation). That, left alone the considerations of performance for storing the names in heap memory.

So far, here's a list of candidates I explored:

1. `lazy_static`. A good candidate, but awful because dependencies.
1. `LazyLock`. Discarded since that's `std`.
1. `SyncUnsafeCell` and `LazyCell`. This is the best, since that's `alloc`, but we need to wait until `SyncUnsafeCell` is no longer a nightly feature.

Reply:
I think the complain was more about registering as an extra constant rather than the type itself.

For the type we could also keep using `[u8; N]` as the type for the field `name` in `TrackInfo` and only use this `StringLike` for the type used in the metadata. So we only use this `StringLike` function with inaccurate metadata (str is not bounded whereas `[u8; N]` is bounded) in the deprecated constant metadata.
Otherwise operations to write or modify the track info will suffer from inaccurate metadata.

For the view function, is it stable already? let's just implement it as an addition. Then the doc for this extra constant should specify it is deprecated in favor of the view and that it can be inaccurate if tracks are stored and modified in storage.

Then all good to me

Reply:
Gotcha! I'll make the change and let you know once it's ready!